src/main.ts

```diff
@@ -8,20 +9,46 @@ import { wait } from "./wait.js"
  */
 export async function run(): Promise<void> {
   try {
-    const ms: string = core.getInput("milliseconds")
+    const versionSpecifiers = parseVersionSpecifiers(
+      core.getInput("versions", {
+        required: true,
+      })
+    )
+    const juliaProject =
+      core.getInput("project", { required: false }) ||
+      process.env.JULIA_PROJECT ||
+      "."
+    const ifMissing = core.getInput("if-missing", { required: false })
 
     // Debug logs are only output if the `ACTIONS_STEP_DEBUG` secret is true
-    core.debug(`Waiting ${ms} milliseconds ...`)
+    core.debug(`versionSpecifiers=${JSON.stringify(versionSpecifiers)}`)
 
-    // Log the current timestamp, wait, then log the new timestamp
-    core.debug(new Date().toTimeString())
-    await wait(parseInt(ms, 10))
-    core.debug(new Date().toTimeString())
+    const resolvedVersions = await resolveVersions(
+      versionSpecifiers,
+      juliaProject,
+      { ifMissing }
+    )
 
-    // Set outputs for other workflow steps to use
-    core.setOutput("time", new Date().toTimeString())
+    const uniqueVersions = versionSort(
+      uniqueArray(resolvedVersions.filter<string>((value) => value !== null))
```

Comment:
Not quite following the scenarios where the version would be null?

Reply:
For example `^2` would currently resolve to `null`. Additionally, we do a HTTP HEAD request to confirm that a nightly version exists. In a scenario where a nightly doesn't exit a `null` would be emitted.